src/app/api/v1/location-autocomplete/route.ts

```diff
@@ -0,0 +1,119 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { NextResponse, NextRequest } from "next/server";
+import uFuzzy from "@leeoniya/ufuzzy";
+
+import { IRelevantLocation } from "../../../../scripts/build/createLocationAutocompleteData/types";
+// The location autocomplete data will be created during the build step.
+// @ts-ignore-next-line
+// eslint-disable-next-line import/no-unresolved
+import locationData from "../../../../../locationAutocompleteData.json";
+
+export type TMatchingLocations = Array<IRelevantLocation> | [];
+
+export interface ISearchLocationParams {
+  searchQuery: string;
+  config: {
+    minQueryLength: number;
+    maxResults: number;
+  };
+}
+
+export interface ISearchLocationResults {
+  searchQuery: string;
+  results: TMatchingLocations;
+}
```

Comment:
Related to above comment about interfaces being prefixed w/ `I`:

<details>
<summary><kbd>git grep -Pho "(export )?interface \w+" | sort | uniq</summary>

```sh
export interface Breach
export interface BreachResolution
export interface BreachResolutionRequest
export interface BreachStats
export interface CircleChartProps
export interface IDataFileUrls
export interface IRelevantLocation
export interface IRelevantLocationAlternate
export interface ISearchLocationParams
export interface ISearchLocationResults
export interface Subscriber
export interface SubscriberBreachResolution
export interface SubscriberEmail
export interface UserBreaches
export interface VerifiedEmail
interface BreachRow
interface EmailAddRequest
interface EmailAddressRow
interface EmailDeleteRequest
interface EmailResendRequest
interface EmailUpdateCommOptionRequest
interface FeatureFlagRow
interface FxaProfile
interface IntrinsicElements
interface JWT
interface JwtPayload
interface PasswordChangeEvent
interface Profile
interface ProfileChangeEvent
interface Session
interface SubscriberRow
interface SubscriptionStateChangeEvent
interface Tables
interface Window
```
</details>

```sh
git grep -Pon "(?:export )?interface [I][A-Z]\w+" -- '*.ts' '*.tsx' | sort | uniq

src/app/api/v1/location-autocomplete/route.ts:16:export interface ISearchLocationParams
src/app/api/v1/location-autocomplete/route.ts:24:export interface ISearchLocationResults
src/scripts/build/createLocationAutocompleteData/types.ts:846:export interface IRelevantLocation
src/scripts/build/createLocationAutocompleteData/types.ts:857:export interface IRelevantLocationAlternate
src/scripts/build/createLocationAutocompleteData/types.ts:867:export interface IDataFileUrls
```

Reply:
Also removed in [e652217](https://github.com/mozilla/blurts-server/pull/3186/commits/e652217a7d9eca0e6b324c2fdca92e6d7dddf49a).